oauth2_http/java/com/google/auth/oauth2/SecureSessionAgent.java

```diff
@@ -210,4 +211,23 @@ private SecureSessionAgentConfig getSecureSessionAgentConfigFromMDS() {
         .setMtlsAddress(mtlsS2AAddress)
         .build();
   }
+
+  private static String validateString(Map<String, Object> map, String key, String errorPrefix)
+      throws IOException {
+    Object value = map.get(S2A_JSON_KEY);
+    if (value == null) {
+      throw new IOException(
+          String.format(OAuth2Utils.VALUE_NOT_FOUND_MESSAGE, errorPrefix, S2A_JSON_KEY));
+    }
+    if (!(value instanceof Map)) {
+      throw new IOException(
+          String.format(OAuth2Utils.VALUE_WRONG_TYPE_MESSAGE, errorPrefix, "Map", S2A_JSON_KEY));
+    }
+    Object address = ((Map<String, Object>) value).get(key);
+    if (!(address instanceof String)) {
+      throw new IOException(
+          String.format(OAuth2Utils.VALUE_WRONG_TYPE_MESSAGE, errorPrefix, "string", key));
+    }
+    return (String) address;
+  }
```

Comment:
If we can extract the above out, can we just revert back to using Oauth2Utils.validateString(...) and remove this method? This implementation looks to be the same as https://github.com/googleapis/google-auth-library-java/blob/cc9826e8636c6162996d83f0f2e98762dd395344/oauth2_http/java/com/google/auth/oauth2/OAuth2Utils.java#L151-L158

Reply:
Yes, it is the same implementation, I extracted out the additional parsing logic into the calling function and reverted back to using `Oauth2Utils.validateString(...)` in the latest commit